pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -335,10 +335,25 @@ def plot_group(keys, values, ax):
             with plt.rc_context(rc):
                 ax = plt.gca()
         data = data._get_numeric_data()
-        if columns is None:
-            columns = data.columns
+
+        if columns:
```

Comment:
Was the condition change here from `is None` to just testing from truthiness required? It would change behavior of `columns=False` which doesn’t look tested; probably best to leave alone